### ISSUE TITLE
Products are synced but do not appear in shop[resolved]; image not fetch in shop

### ISSUE THREAD

**Issue by cmw09** [facebookincubator/facebook-for-woocommerce#463]
Please help!

Link to my shop:https://www.facebook.com/pg/cherrymayward/shop/

I came across this issue in the catalog manager: 

"Events
Missing "AddToCart, Purchase, ViewContent" events in Cherry Ward's Pixel
We haven't detected the events AddToCart, Purchase, ViewContent coming from your pixel"

**Comment by mengyingdu** [comment]
Hi @cmw09 ,

What other plugin are you using? I found different pixel event which shouldn't fired by our plugin on your site.

Thanks.

**Comment by waywardchickadee** [comment]
I'm having the same problem:.  Here is the All Products page:
https://www.facebook.com/pg/waywardchickadee/shop/?cid=197321304404148

And there are several products in the catalog that don't appear in the shop, for example:
https://www.facebook.com/commerce/products/2083279321742255/
https://www.facebook.com/commerce/products/1676127485804970/

And the catalog says there is an error:
![image](https://user-images.githubusercontent.com/38990171/39647722-106a0bb0-4fae-11e8-8b7a-8512d96dd1b0.png)

But when I go to the page it is firing correctly. 

I am using Facebook for WooCommerce Version 1.8.7 and Wordpress version 4.9.5 and WooCommerce Version 3.3.5. I'm not using any other facebook plugins.

**Comment by cmw09** [comment]
@mengyingdu I just tried to use Pixel Caffeine a few hours after I couldn't get this to work.. Should I delete it?

**Comment by blieven1** [comment]
Same issue here: www.facebook.com/lagomcocoon

Upload succeeded but no products visible in te shop, only in the catalog.

**Comment by mengyingdu** [comment]
Hi @cmw09  @blieven1 @waywardchickadee ,

Huge thanks for your reports! This is a bug on Facebook side and from my triangle, it raise a SEV which a couple of engineers working on it now.

We have fixed it and is in pushing stage from 10 PM PST yesterday. You will expect to see your products 
on page later today.


Again, thanks all!

**Comment by mengyingdu** [comment]
Instructions to delete settings and re-setup:

1. go to 'woocommerce' -> settings  -> integrations -> facebook-for-woocommerce
2. In the blue panel, click on 're-config'
3. In the popup window, 'advance option' -> delete your settings
4. back to the blue panel, click on 'get started' -> finish setup

**Comment by blieven1** [comment]
Thanx! It's indeed solved, all products are available but some don't show there images on the shop page now. Images are available in the catalog, another fb bug?

**Comment by cmw09** [comment]
Thank you so much, this solved it :)


www.cherrymayward.com
hello@cherrymayward.com
facebook.com/cherrymayward
www.cherrymayward-blog.com

**Comment by mengyingdu** [comment]
Hi @blieven1 ,

You may need to wait longer time to show all of your images after syncing products. That's because we use async to update images in order to make sure the critical product info sync between client DB and FB db.

Can you check your images later today (after a couple of hours) ?

Thanks for your patience.

**Comment by blieven1** [comment]
Hi, @mengyingdu 
It's 15hrs later but they still don't show up in the shop page. Is this normal?
I do see them in instagram product tagging so upload seems to be ok.

Thanx,
Bjorn

**Comment by mengyingdu** [comment]
Hi @blieven1 ,

1. I checked your fb shop, there are 2 products without images. 

Is it the same when you check it? If there are more showing up than before, it should be fine. 
![screen shot 2018-05-08 at 9 15 15 am](https://user-images.githubusercontent.com/24535640/39769240-7d8cb540-52a0-11e8-8514-f9ad75e49b4f.png)

2. I checked your catalog, those TWO products on your shop without images actually have image url in our database. I think your products synced no problems.

Thanks.

**Comment by blieven1** [comment]
Hi, yes it’s the same 2 that don’t show up since the upload. There are also other products where there is one or more pictures missing. At least they have one so this isn’t a big deal. But no pictures are not nice for customers.

Yes the catalog and upload seems to be completly correct. So probably another fb bug itself and not the plugin.

Can you forward the issue to someone there?

Thanx!

**Comment by mengyingdu** [comment]
Hi @blieven1 ,

I flag your problems to shop team and catalog team. It turns out that your products images have been sync to FB DB successfully. However, there's sth wrong when the page tried to fetch the image. We are not sure the total number of users being affected by this issue, but we are discussing how to solve it now. 

Thanks.

**Comment by cmw09** [comment]
I have this issue as well... I've added another image to one of my products
and it doesn't show up on fb


www.cherrymayward.com
hello@cherrymayward.com
facebook.com/cherrymayward
www.cherrymayward-blog.com

**Comment by blieven1** [comment]
@mengyingdu thanx for confirming that it is a bug, any news about a fix?

Thanx,
Bjorn

**Comment by mengyingdu** [comment]
Hi @blieven1 and @cmw09 ,

I have flagged this to shop team and wait for update now. Basically, there are some products fail to render to images when loading the page. It's not a bug from plugin.

Thanks for your patience.

**Comment by mengyingdu** [comment]
Hi @cmw09 ,

I can see every product in your shop has products now.

**Comment by mengyingdu** [comment]
@blieven1 ,
Yes, i confirmed that there are some products without images in your shop.

**Comment by blieven1** [comment]
Hi,

Any news about a fix from the shop team? It's been a while now.

Thanx,
Bjorn

**Comment by mengyingdu** [comment]
Hi @blieven1 ,

I looked at one of the items with failed image fetch, it seems like the image exceeded our maximum supported file size. Can you try a different image or resize it?

Thanks.

**Comment by blieven1** [comment]
Hi, 

That was it, some of my images are over 5000px wide. Resized them to 3000px and it worked.
Case closed ;)
Thanx,
Bjorn

**Comment by mengyingdu** [comment]
All issues in this thread are fixed. Thanks all.

**Comment by fhuacuja** [comment]
My woocommerce products are not appearing in my Facebook's shop page. They are synced,
I can only see them here: https://www.facebook.com/products/catalogs/2405547306184014/products?nav_source=dynamic_instant_ads

**Comment by Spezaa** [comment]
Hi I have the same problem my website is donhardware.com I have synced the products but they are not showing in facebook shop as you can see here https://www.facebook.com/pg/DonHardwareOfficial/shop

**Comment by panda01** [comment]
What is the maximum image size?

**Comment by panda01** [comment]
@mengyingdu

**Comment by alcantech** [comment]
I Solve my problem by uninstalling the shop on my Facebook Page, don't worry only the FB shop page will reset. And uninstalling and resetting the WordPress plugins to Setup mode. On the second attempt, Do not set up shop or agreeing with any terms as shown on youtube. All you have to do is let the plugin do all the works when done check your shop page in your Fb page or check your catalog on your FB Business Manager. I hope this works also with others.

**Comment by iamanderick** [comment]
If anyone is still having this same problem in 2020, this how I fixed it for a client. https://www.facebook.com/pg/queennette.fashion/shop/

Just download the latest stable version 1.9.1 from here.. From the time you might be reading this fix, you may have downloaded the version 1.9.15+. Just delete it from your WordPress dashboard and download the version from the link above.

Then go to WooCommerce > Integration > Facebook for WooCommerce > Settings.
In the Pop window that appears, go to Advanced Options and choose Remove. When you finish removing it, try reconnecting back again. That should fix the problem now.

In Summary;

Downgrade to version 1.9.1
Remove the old settings and do a new one. That's all! Cheer!

**Comment by lizleeds** [comment]
Can anyone help with images that do not show up even after, reinstalling and reconfiguring woocommerce facebook plugin, also I have changed px of images to max 530 px, still donot know what to do, I will wait one h to see does it work.
https://www.facebook.com/pg/SqueakyCleanup.co.uk/shop/?ref=page_internal

**Comment by lizleeds** [comment]
![product](https://user-images.githubusercontent.com/62110885/76541916-30764100-647c-11ea-9c65-ed5d7472baa1.PNG)
ok i do try to add new product, it asked for link to existing product so i should have two first products duplicated, it shows processing.. will I have to add manually all products??

**Comment by lizleeds** [comment]
oook. so no one helped but now I know what to do, there are two methods:
1. you add all products one by one or in bulk (if you have a file) but to each product you need to paste link to that product from your own website shop
2. after sync you go to each product and replace a picture that should be max 530x530
I prefer way number 2 and I can keep better resolution pictures in my squeakycleanup shop.

if this helps anyone I am  glad, took me one day to find out but no one is writing about this anywhere. They could repair this issue or at least write a guide in your fb shop to not waste your time solving why images do not show.

**Comment by lizleeds** [comment]
![1](https://user-images.githubusercontent.com/62110885/76632099-90cbb800-653a-11ea-91e3-87db525a5608.PNG)
![2](https://user-images.githubusercontent.com/62110885/76632100-92957b80-653a-11ea-8e61-ea7119c94244.PNG)
![3](https://user-images.githubusercontent.com/62110885/76632112-96290280-653a-11ea-9451-e35926733f14.PNG)
![4](https://user-images.githubusercontent.com/62110885/76632122-9923f300-653a-11ea-8eb2-86a9981900a0.PNG)
![5](https://user-images.githubusercontent.com/62110885/76632124-9aedb680-653a-11ea-84aa-3fa4505cdbf1.PNG)

So I uploaded method number 2.
In red i do show what link was included during synchronisation so I did not have to do method number 1. But if I would have issues with sync and would have to add each product maual -you will be required to add your own shop link. (possible not if you set in settings that you do not have online shop)

**Comment by lizleeds** [comment]
ps. you do not have to change image dimentions works well with my larger pictures

**Comment by lizleeds** [comment]
![0](https://user-images.githubusercontent.com/62110885/76632848-bf965e00-653b-11ea-8aa7-c6cf57456ebf.PNG)
![11](https://user-images.githubusercontent.com/62110885/76632853-c329e500-653b-11ea-9eae-208d66e822c1.PNG)
